File: Tema 10/ListadoPersonasClean/Domain/Interfaces/IPersonaUseCase.cs
using Domain.DTOS;
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    public interface IPersonaUseCase
    {
           
        List<PersonaConNombreDepartamento> getPersonas();

        PersonaConNombreDepartamento getPersona(int id);

        PersonaConListadoDepartamento getPersonaEditar(int id);

        PersonaConListadoDepartamento getPersonaAgregar();

        List<Departamento> getDepartamentos();

        int agregarPersona(Persona persona);

        int actualizarPersona(Persona persona);

        int eliminarPersona(int id);
    }
}

File: Tema15/ListadoPersonasAngular/src/Domain/Interfaces/IPersonaUseCase.ts
import { PersonaDTO } from "../DTOs/PersonaDTO.js";

export interface IPersonaUseCase {
  getPersonas(): Promise<PersonaDTO[]>;
  getPersona(id: number): Promise<PersonaDTO>;
  agregarPersona(persona: PersonaDTO): Promise<number>;
  actualizarPersona(persona: PersonaDTO): Promise<number>;
  eliminarPersona(id: number): Promise<number>;
}